src/Transform.jl

```diff
@@ -40,17 +33,14 @@ function logit{NumType <: Number}(x::NumType)
 end
 
 
-function logit{NumType <: Number}(x::Array{NumType})
-    1.0 ./ (1.0 + exp(-x))
-end
-
-
 """
 Convert an (n - 1)-vector of real numbers to an n-vector on the simplex, where
 the last entry implicitly has the untransformed value 1.
 """
 function constrain_to_simplex{NumType <: Number}(x::Vector{NumType})
     if any(x .== Inf)
+        # Is there a bug here? If more than 1 entry in x is Inf, that just means
+        # that the last entry in z is 0, not the both entries in z are the same.
```

Comment:
I think you are proposing a pre-filtering step that maps unbounded floats into some box of legal values.  That is the kind of thing that causes many optimization routines to crash (e.g. BFGS) since it effectively produces zero gradients where the routine expects nonzero gradients.  It says that the ELBO is constant outside a certain range with a derivative discontinuity at the edge of the box.  If you want to give that a try, why not do it in a separate CL and test on a bunch of objects?

Here's my reasoning about the current logic.  When an optimization routine passes a pair of ```Inf````s, what should Celeste do?  I think it should gracefully choose a legal constrained parameter value, pass it to the optimizer, and let the model say that it is an extremely bad set of parameters.  That is what this does (and what it has been doing, just fine, for many months.

One potential problem with ```Inf``` is that finite gradient steps cannot get away from them.  If we find that this is a real problem, it seems to me that there are more graceful solutions than a bounding box (like letting the rest of the model be fit and then restarting with finite values).  I would probably cross this bridge when we come to it, but if you feel like working on it now, feel free.